### ISSUE TITLE
Block illegal moves on check condition doesn't works

### ISSUE THREAD

**Issue by hpsaturn** [hpsaturn/esp32-chess-engine#1]
## Summary

The code works fine but I found one issue, when you are in check situation you able to move other pieces (illegal move) and the code not block it. In order to fix this issue I decided migrate the code to better environment for discussion and report bugs or issues, here in Github, and also I did some changes for have it in better platform for compiling, development, debugging, etc, PlatformIO with Arduino framework too.

**Comment by bleckett** [comment]
Hey hpsaturn - any new updates on this engine or is it dead? I noticed that you can just ignore the checks and move whatever you want. You can even get your king eaten and you can still move other pieces... LOL.

**Comment by hpsaturn** [comment]
Yes, I think that it is dead. I'm going to share it, I hope that someone help us. I don't have time.

**Comment by bleckett** [comment]
Hi again... I ended up writing my own function to check if (white's) move is valid (king in check/checkmate) before before passing it to the engine. I know it's probably repetitive, but I wanted to use it for the chessboard I was making. I'll see about adding it here once I get a little more confident in using GitHub. PS - you can chessboard making process here if you're interested...
https://www.youtube.com/watch?v=5S0z9_bPXYM

**Comment by hpsaturn** [comment]
Nice catch! Beautiful creation! Congrats, nice job man! 

Well, with Github is easy make a "pull request" 

In general is only the next steps:
1. Fork this repo
2. In your fork using the Github editor, edit the target file and add your lines or changes.
3. On Github too make a commit (save your changes in your fork)
4. After that in your fork home, the **pull request** button should be enable
5. make the pull request and that its.

More info here:

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork

Video:
https://www.youtube.com/watch?v=l5NrYIa_aG4

Of course, also you can do it with the command line.
